pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseCombineOperator.java

```diff
@@ -118,6 +119,9 @@ public void runJob() {
     IntermediateResultsBlock mergedBlock;
     try {
       mergedBlock = mergeResults();
+    } catch (InterruptedException ie) {
+      Thread.currentThread().interrupt();
```

Comment:
We don't want to interrupt the thread again since the interrupt is already handled. Same for other places

Reply:
can do, if we're sure the interrupt status is not needed by following logic.